jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/skaffold/SkaffoldWatchParameters.java

```diff
@@ -17,35 +17,34 @@
 package com.google.cloud.tools.jib.gradle.skaffold;
 
 import java.io.File;
-import java.nio.file.Path;
-import java.util.Collections;
-import java.util.Set;
-import java.util.stream.Collectors;
 import javax.inject.Inject;
 import org.gradle.api.Project;
-import org.gradle.api.tasks.Internal;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.tasks.InputFiles;
 
 /** Skaffold specific JibExtension parameters for configuring files to watch. */
 public class SkaffoldWatchParameters {
 
-  private final Project project;
-
-  private Set<Path> buildIncludes = Collections.emptySet();
-  private Set<Path> includes = Collections.emptySet();
-  private Set<Path> excludes = Collections.emptySet();
+  private final ConfigurableFileCollection fileCollection;
+  private ConfigurableFileCollection buildIncludes;
+  private ConfigurableFileCollection includes;
+  private ConfigurableFileCollection excludes;
 
   @Inject
   public SkaffoldWatchParameters(Project project) {
-    this.project = project;
+    this.fileCollection = project.getObjects().fileCollection();
+    this.buildIncludes = project.getObjects().fileCollection();
+    this.includes = project.getObjects().fileCollection();
+    this.excludes = project.getObjects().fileCollection();
   }
 
   /**
    * A set of absolute paths to include with skaffold watching.
    *
    * @return a set of absolute paths
    */
-  @Internal
-  public Set<Path> getBuildIncludes() {
+  @InputFiles
```

Comment:
I don't know how changing these from `@Internal` to `@InputFiles` affects the UP-TO-DATE-ness of these skaffold tasks, but these tasks are required to always execute, so we don't want them to be every up-to-date

Reply:
Are you sure? From the docs on @InputFiles: "task to be considered out-of-date when the file paths or **contents have changed**"; so my reasoning here was: if skaffold changed either the paths or the files themselves have changed, then run it, otherwise our inputs are unchanged so no need to run it

Reply:
This is a task called by skaffold to understand which paths to watch when using a jib build. It's data scaffold wants to set it's own watcher. Called from the skaffold process.

Reply:
Ok, reverted back to @Internal